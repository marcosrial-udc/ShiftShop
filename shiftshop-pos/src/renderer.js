import * as React from 'react';
import * as ReactDOM from 'react-dom';
import {Provider} from 'react-redux';
import {MuiThemeProvider} from '@material-ui/core';
import {SnackbarProvider} from 'notistack';
import {IntlProvider} from 'react-intl';

/* Async/await regenerator necesaries */
import 'core-js/stable';
import 'regenerator-runtime/runtime';

import './assets/scss/index.scss';
import theme from './theme'

import {App} from './modules/app';

import configureStore from './store';
import {initReactIntl} from "./i18n";
import './polyfills';

/* Configure store. */
const store = configureStore();

/* Configure i18n. */
const {locale, messages} = initReactIntl();

ReactDOM.render(
    <MuiThemeProvider theme={theme}>
      <Provider store={store}>
        <IntlProvider locale={locale} messages={messages}>
          <SnackbarProvider anchorOrigin={{vertical: 'top', horizontal: 'center'}}
                            autoHideDuration={2500}>
            <App/>
          </SnackbarProvider>
        </IntlProvider>
      </Provider>
    </MuiThemeProvider>,
    document.getElementById("root"));
