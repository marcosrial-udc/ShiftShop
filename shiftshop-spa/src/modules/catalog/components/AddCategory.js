import React, {useEffect, useMemo, useRef, useState} from 'react';
import {useDispatch} from 'react-redux';
import {FormattedMessage} from 'react-intl';
import {useSnackbar} from 'notistack';
import {
    Box,
    Button,
    CircularProgress,
    Dialog,
    DialogActions,
    DialogContent,
    DialogTitle,
    TextField,
    Typography
} from '@material-ui/core';
import {Add, AddBox} from '@material-ui/icons';

import useStyles from '../styles/AddCategory';

import * as actions from '../actions';
import {formValidator} from '../../../utils';
import {Alert, ErrorContent} from '../../common';

const AddCategory = () => {
    const classes = useStyles();

    const dispatch = useDispatch();
    const {enqueueSnackbar, closeSnackbar} = useSnackbar();

    const _isMounted = useRef(true);
    useEffect(() => {
        return () => _isMounted.current = false
    }, []);

    const [name, setName] = useState('');
    const [adding, setAdding] = useState(false);
    const [errors, setErrors] = useState(null);
    const [open, setOpen] = useState(false);

    const handleChangeName = event => {
        setName(event.target.value);
    };

    const closeErrors = () => setErrors(null);
    const openDialog = () => setOpen(true);
    const closeDialog = () => setOpen(false);

    const isValidName = useMemo(() => formValidator.isNotEmpty(name), [name]);

    const handleSubmit = e => {

        e.preventDefault();
        closeErrors();

        if (isValidName && !adding) {
            addCategory();
        }
    };

    const addCategory = () => {

        if (adding) return;

        setAdding(true);

        const categoryName = name.trim();

        dispatch(actions.addCategory({name: categoryName},
                _ => {
                enqueueSnackbar(<FormattedMessage id="project.catalog.AddCategory.success"/>,
                    {
                        variant: 'success',
                        autoHideDuration: 1500,
                    });
                _isMounted.current && setName('');
            },
            errors => {
                if (_isMounted.current) {
                    setErrors(errors)
                } else {
                    enqueueSnackbar(<ErrorContent errors={errors}/>,
                        {
                            variant: 'error',
                            persist: 'true',
                            action: key => (<Button color="inherit" variant="outlined" onClick={() => closeSnackbar(key)}>
                                <FormattedMessage id="project.global.button.close"/>
                            </Button>)
                        });
                }
            },
            () => _isMounted.current && setAdding(false)));

    };

    return (
        <div>
            <Button color="primary" variant="contained" size="small" onClick={openDialog}>
                <Add/> <FormattedMessage id="project.catalog.AddCategory.addButton"/>
            </Button>
            <Dialog className={classes.root} maxWidth="xs" fullWidth open={open} onClose={closeDialog}>
                <DialogTitle>
                    <Box display="flex" alignItems="center">
                        <AddBox fontSize="default" color="primary"/>&nbsp;
                        <Typography variant="h5">
                            <FormattedMessage id="project.catalog.AddCategory.title"/>
                        </Typography>
                    </Box>
                </DialogTitle>
                <form onSubmit={e => handleSubmit(e)} noValidate>
                    <DialogContent dividers>
                        {errors && <Alert className={classes.alert} message={errors} backendError onClose={closeErrors}
                                          variant="error"/>}
                        <TextField margin="dense" variant="outlined" fullWidth autoFocus disabled={adding}
                                   label={<FormattedMessage id="project.global.field.name"/>} value={name}
                                   onChange={handleChangeName} required/>
                    </DialogContent>
                    <DialogActions>
                        <Button variant="contained" color="secondary" onClick={closeDialog} disableElevation>
                            <FormattedMessage id="project.global.button.close"/>
                        </Button>
                        <Button variant="contained" color="primary" type="submit" disabled={!isValidName || adding} disableElevation>
                            {adding && <CircularProgress className={classes.buttonProgress} size={24}/>}
                            <FormattedMessage id="project.global.button.add"/>
                        </Button>
                    </DialogActions>
                </form>
            </Dialog>
        </div>
    );
};

export default AddCategory;
