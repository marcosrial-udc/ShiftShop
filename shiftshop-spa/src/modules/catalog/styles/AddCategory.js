import {makeStyles} from '@material-ui/core';

const useStyles = makeStyles(theme => ({
    root: {padding: theme.spacing(3)},
    alert: {marginBottom: theme.spacing(2)},
    buttonProgress: {position: 'absolute'},
}));

export default useStyles;
