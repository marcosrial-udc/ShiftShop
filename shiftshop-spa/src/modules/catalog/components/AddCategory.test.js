import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import {Provider} from 'react-redux';
import {IntlProvider} from 'react-intl';
import {SnackbarProvider} from 'notistack';

import {initReactIntl} from "../../../i18n";

import AddCategory from './AddCategory';

const {messages} = initReactIntl();

describe('AddCategory snapshot', () => {

    const middlewares = [];
    const mockStore = configureStore(middlewares);

    const createAddCategory = store => {

        return renderer.create(
            <SnackbarProvider>
                <Provider store={store}>
                    <IntlProvider locale='en' messages={messages}>
                        <AddCategory/>
                    </IntlProvider>
                </Provider>
            </SnackbarProvider>
        );

    };

    test('Initial state', () => {

        // Initialize mockstore with empty state
        const initialState = {};
        const store = mockStore(initialState);

        const addCategory = createAddCategory(store);

        expect(addCategory.toJSON()).toMatchSnapshot();

    });

});
