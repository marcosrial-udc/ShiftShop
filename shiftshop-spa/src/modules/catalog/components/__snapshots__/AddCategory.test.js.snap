// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCategory snapshot Initial state 1`] = `
<div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
        />
      </svg>
       
      New
    </span>
  </button>
</div>
`;
